### PR TITLE
Add suggested web cache key for remove accounts

### PR DESCRIPTION
**Pull request by neha-bhargava**
Fixes # .
#2643 

**Changes proposed in this request**
Add suggested web cache key for remove accounts call.

**Testing**
Added unit test

**Performance impact**
None

### CODE REVIEW

**Review by jmprieur — APPROVED** [review]
LGTM.
Thanks @neha-bhargava 

@abhidnya13 : do you want to have a look as well?

File: tests/Microsoft.Identity.Test.Unit/CacheTests/SuggestedWebCacheKeyTests.cs [hunk 1]
```diff
@@ -86,5 +86,40 @@ public void TestCacheKeyForTenantAuthority()
             var expectedKey = $"{this._serviceBundle.Config.ClientId}_{tenantId}_AppTokenCache";
             Assert.AreEqual(expectedKey, actualKey);
         }
+
+        [TestMethod]
+        public void TestCacheKeyForRemoveAccount()
+        {
+            // Arrange
+            const string tenantId = TestConstants.AadTenantId;
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Test.Unit/CacheTests/SuggestedWebCacheKeyTests.cs:94]: >tenantId [](http://example.com/codeflow?start=25&length=8)

nit: don't need a variable, as it's only used once.

File: tests/Microsoft.Identity.Test.Unit/CacheTests/SuggestedWebCacheKeyTests.cs [hunk 2]
```diff
@@ -86,5 +86,40 @@ public void TestCacheKeyForTenantAuthority()
...
+            var appTokenCache = new TokenCache(this._serviceBundle, isApplicationTokenCache: true);
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Test.Unit/CacheTests/SuggestedWebCacheKeyTests.cs:95]: >this [](http://example.com/codeflow?start=47&length=4)

nit: remove `this.` in all the places it's used.

File: tests/Microsoft.Identity.Test.Unit/CacheTests/SuggestedWebCacheKeyTests.cs [hunk 3]
```diff
@@ -86,5 +86,40 @@ public void TestCacheKeyForTenantAuthority()
...
+            var requestContext = new RequestContext(this._serviceBundle, Guid.NewGuid());
+            var tenantAuthority = AuthorityInfo.FromAadAuthority(AzureCloudInstance.AzurePublic, tenant: tenantId, validateAuthority: false);
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Test.Unit/CacheTests/SuggestedWebCacheKeyTests.cs:97]: nit: no need for variable. pass it directly.

File: tests/Microsoft.Identity.Test.Unit/CacheTests/SuggestedWebCacheKeyTests.cs [hunk 4]
```diff
@@ -86,5 +86,40 @@ public void TestCacheKeyForTenantAuthority()
...
+            };
+
+
+            // Act
+            var actualKey = SuggestedWebCacheKeyFactory.GetKeyFromRequest(parameters);
+
+            // Assert
+            Assert.IsNotNull(actualKey);
+            var expectedKey = parameters.HomeAccountId;
+            Assert.AreEqual(expectedKey, actualKey);
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Test.Unit/CacheTests/SuggestedWebCacheKeyTests.cs:122]: nit: no variable

**Comment by jennyf19** [comment]
can we have an integration test that uses it e2e and removes w/the cache key?

---
In reply to: [700360728](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2745#issuecomment-700360728) [](http://example.com/codeflow?ancestors=700360728,700360728)

**Review by jennyf19 — CHANGES_REQUESTED** [review]
:clock1:

**Review by pmaytak — APPROVED** [review]

**Comment by trwalke** [comment]
> using Microsoft.Identity.Client;

This file is missing copywright info

---
Refers to: tests/Microsoft.Identity.Test.Unit/CacheTests/SuggestedWebCacheKeyTests.cs:2 in 5800554. [](commit_id = 5800554751339bd30e5f07e36ac151e306ab2565, deletion_comment = False)

**Review by trwalke — APPROVED** [review]
LGTM. Just need copyright info

**Comment by bgavrilMS** [comment]
@jennyf19  - I updated MSAL's integration test around GetTokenByAuthCode with this. It's the only confidential client scenario where RemoveAccount can be used.

**Review by jennyf19 — APPROVED** [review]

**Pull request merged**